core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -424,9 +424,34 @@ private[chisel3] class DynamicContext(
   var inDefinition: Boolean = false
 
   // Used to compute minimally relative source info paths
-  val sourcePathComputor: SourcePathComputor = SourcePathComputor.empty()
+  val sourcePathComputor: SourcePathComputor = {
+    val computor = SourcePathComputor.empty()
+    val paths = SourceFiles.getSourceFilePaths()
+    paths.foreach { p => computor.addPath(p) }
+    computor
+  }
+}
+
+object SourceFiles {
+  import java.io.File
+  def getSourceFilePaths(): List[Seq[String]] = {
+    def rec(file: File): List[Seq[String]] = {
```

Comment:
Have you considered using `os-lib` instead of the `java.io` package? Some of the functions in the `os` package are much nicer to work with and most new code in `firrtl` and `chiseltest` has made use of them.

Reply:
In this case you would work with a `os.Path` instead of a `File`

Reply:
This isn't needed anymore with relying on the "user-dir" system property.